src/rust/engine/src/rule_graph.rs

```diff
@@ -290,6 +306,9 @@ impl <'t> GraphMaker<'t> {
                                    rules_or_literals_for_selector);
               },
               &Selector::SelectDependencies(SelectDependencies{ref product, ref dep_product, ref field_types, ..}) |
+              // TODO This is incorrect because SelectTransitive works differently than SelectDependencies.
+              // See line nodes.rs:710, select transitive depends on itself w/ a different subject
+              // // TODO TODO I think ^^ the above todo is incorrect. Need to think it through again.
```

Comment:
I kept getting distracted while looking at this. I'll go over it one more time and rm the comment.